core/src/Kokkos_Macros.hpp

```diff
@@ -116,7 +116,11 @@
 // Code is parsed and separated into host and device code.
 // Host code is compiled again with another compiler.
 // Device code is compile to 'ptx'.
-#define KOKKOS_COMPILER_NVCC __NVCC__
+// NOTE: There is no __CUDACC_VER_PATCH__ officially, its __CUDACC_VER_BUILD__
+// which does have more than one digit (potentially undefined number of them).
+// This macro definition is in line with our other compiler defs
+#define KOKKOS_COMPILER_NVCC \
+  __CUDACC_VER_MAJOR__ * 100 + __CUDACC_VER_MINOR__ * 10
```

Comment:
Is there a good reason to not define this as `CUDA_VERSION`?

Reply:
This is about which version of NVCC this is independent of whether we compile with clang or nvc++. Furthermore, I thought that KOKKOS_COMPILER_FOO is for most other compilers defined as FOO_VER_MAJOR*100+FOO_VER_MINOR*10+FOO_VER_PATCH.
Essentially there is a difference between checking for CUDA version and compiler version for the compiler you use to compile CUDA. Just for NVCC its kinda the same (except our fuckup with being inconsistent with KOKKOS_COMPILER_CUDA_VERSION)

Reply:
Yeah, we should probably try to remove `KOKKOS_COMPILER_CUDA_VERSION`.